Fix incorrect use of BN_CTX API

agustingianni opened the pull request:
In some edge cases BN_CTX_end was being called without first calling
BN_CTX_start. This creates a situation where the state of the big
number allocator is corrupted and may lead to crashes.

Fixes #13812

mattcaswell reviewed (approved):
I think this may stray beyond our definition of trivial (which is generally reserved for very simple fixes, and typos etc). Can you submit a CLA?

https://www.openssl.org/policies/cla.html

t8m commented:
I agree with Matt that this is beyond the trivial limit.

So please remove the CLA: trivial from the commit message and please also add the Fixes line to the commit message instead.

agustingianni commented:
Hello @mattcaswell and @t8m as requested I have signed and sent the CLA to the openssl foundation. I have also removed the trivial tag and added the fixes line to the commit message.

Lemme know if there is anything else I need to do.

Thank you!

mattcaswell commented:
Close/reopen to kick CLA bot

mattcaswell commented:
CLA received - thank you! My approval stands.

Needs a second review.

openssl-machine commented:
This pull request is ready to merge

t8m commented:
Merged to master. Thank you for the contribution?

As this is something that should be fixed on the 1.1.1 branch but the cherry pick will not apply cleanly, would you like to submit a PR against the 1.1.1 branch, please?
